### DISCUSSION TITLE
关于免费SSL证书自动申请和续期功能怎么使用问题

### DISCUSSION THREAD

**Discussion by nh-199** [Safe3/uusec-waf#57] [Q&A] [answered]
信息描述：本人使用的4.1.0版本，在证书管理界面只找到添加证书=》填写或上传两个功能，没有找到自动续签的设置，有哪个兄弟知道怎么使用吗？
![image](https://github.com/Safe3/uuWAF/assets/25094463/73ae582c-3979-49db-ad49-476a7db73b68)
![image](https://github.com/Safe3/uuWAF/assets/25094463/541bdea2-3ec8-4899-a4bd-48e737e35aeb)
![image](https://github.com/Safe3/uuWAF/assets/25094463/353ee0b1-464d-4868-a443-4c0c706b4b03)
功能添加版本：v.4.0.1
![image](https://github.com/Safe3/uuWAF/assets/25094463/2e1ff9ff-edd6-4d05-ab32-f3befe19df3d)

**Comment by Safe3** [comment, marked as answer]
南墙采用无感SSL申请续期功能：

只需将对应域名dns指向南墙，然后访问https:// 域名，南墙会自动进行SSL证书申请和续期，证书默认保存在/uuwaf/web/certificates目录。

**Reply by nh-199** [reply]
是不用填写或上传证书，只需将对应域名dns指向南墙，然后访问https:// 域名后，就会自动进行SSL证书申请和续期吗？
我这边使用的CloudFlare DNS指向南墙做了次实现没有成功，不清楚是不是对CF这边支持有问题？

**Reply by Safe3** [reply]
默认使用的let encrypt证书，要是你之前已经申请的比较多，也可能造成不能申请，具体可以看看syslog日志或南墙/uuwaf/logs/error.log，里面有acme申请记录

**Reply by nh-199** [reply]
好的，我这边检查到了acme的申请错误信息（`Could not find solver for: tls-alpn-01`），是我的环境问题，十分感谢！

**Reply by Safe3** [reply]
可以发完整点的信息看看

**Reply by nh-199** [reply]
1panel环境中安装的uuwaf，完整的日志信息是：
![image](https://github.com/Safe3/uuWAF/assets/25094463/715836ec-1c21-4a1c-98d0-021e086b1fbf)

**Reply by Safe3** [reply]
真正的原因还在下面几行，你发的图片看不到

**Reply by nh-199** [reply]
好的，之前一直在retry没有看到新的信息，经过提示，重新看了下最下面出来新的信息了
![image](https://github.com/Safe3/uuWAF/assets/25094463/01654151-216b-4f4e-b0a6-58f2b60c2f73)
这个信息里面的地址是我的旁路由地址，我的1panel所在虚拟机设置的是主路由IP地址作为网关地址，不清楚什么原因会爆出来这个地址，我去检查下我的网络环境，感谢帮忙查看！！！
